lib/spack/spack/solver/asp.py

```diff
@@ -1759,13 +1759,30 @@ def key_fn(item):
             # All the preferred version from packages.yaml, versions in external
             # specs will be computed later
             version_preferences = packages_yaml.get(pkg_name, {}).get("version", [])
-            for idx, v in enumerate(version_preferences):
-                # v can be a string so force it into an actual version for comparisons
-                ver = vn.Version(v)
+            version_defs = []
+            pkg_class = spack.repo.path.get_pkg_class(pkg_name)
+            for vstr in version_preferences:
+                v = vn.ver(vstr)
+                if isinstance(v, vn.GitVersion):
```

Comment:
do we support `version("ref=version")`?

Reply:
In a `package.py`? I don't actually know: would something go wrong if we did? I don't think we enforce allowing it in tests, and I don't see any mock packages which do that (`grep -i "version[^,]*\=" -r var/spack/repos/builtin.mock/packages/`). The way we suggest (and how all packages in `builtin` do it) is:

```
version("4.0.1-xsdk", commit="c55c80d17b82d80de04b849dd526e17044f8c99a")
```

Reply:
Yeah, shouldn't be an issue also if there was a GitVersion that could've satisfied smth from pkg_class.versions